website/docs/pipelines/promoting-applications.mdx

```diff
@@ -130,133 +133,151 @@ Be sure to create the Provider/Alert tuple on **each of the leaf clusters
+3. Prefer granting access to specific secrets, rather than granting `list` and `watch` on secrets as:
 
-### Credentials Secret
+> It is also important to note that list and watch access also effectively allow for users to reveal the Secret contents.
 
-In the journey of creating a pull request, there are different secrets involved:
+#### Policy
 
-1. Pipeline controller receives events via [webhook from leaf clusters](./#setup-notifications-from-leaf-clusters). Hmac is used for authN/authZ so an hmac key should be provided in this case.
-2. Pipeline controller clones and patches manifests to promote from the pipeline configuration repo. A set of [git credentials](https://fluxcd.io/flux/components/source/gitrepositories/#secret-reference) are required.
-3. Pipeline controller uses git provider api to create the pull request with the promoted manifests. A Personal Access Token (PAT) needs to be created to interact with pipelines git provider API. This PAT is also used to list pull requests from the configured repository.
+By following the guidelines above, you can have a safe initial configuration. However, given there are no deny semantics in [RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole), you need to guard future changes.
 
-Create a Kubernetes secret with the previous data.
+> An RBAC Role or ClusterRole contains rules that represent a set of permissions.
+> Permissions are purely additive (there are no "deny" rules).
 
-<details><summary>Expand to see example</summary>
+You should ensure that attempts to break this contract are blocked and detected. You could achieve it
+by using Weave GitOps' [Policy capabilities](../../policy/intro/). Policy Agent acts in two complementary modes:
+- [Admission Controller](../../policy/intro#admission-controller) protects from any attempt to create non-compliant RBAC resources that would end granting access to the secret.
+- [Audit](../../policy/intro#audit) helps you identify already existing resources that are out of compliance. For example,
+roles created before policy agent was introduced as admission controller.
 
-```shell
-# example to use git over https with basic auth and pat
-$ kubectl create secret generic promotion-credentials \
-  --namespace=pipeline-01 \
-  --from-literal="username=<bot account name>" \
-  --from-literal="password=<token value>" \
-  --from-literal="token=<token value>" \
-  --from-literal="hmac-key=<hmac-key value>"
-```
+Once you have enabled Policy, the [Policy Library](../../policy/weave-policy-profile/) gives you a set of
+good practices policies that will help you securing your pipeline secrets according to previous RBAC recommendations. Deploy
+them as Kustomization based on the following example:
 
 ```yaml
+apiVersion: source.toolkit.fluxcd.io/v1
+kind: GitRepository
+metadata:
+  name: policy-library
+  namespace: flux-system
+spec:
+  interval: 10m0s
+  url: https://github.com/weaveworks/policy-library.git
+  secretRef:
```

Comment:
review with @MostafaMegahid whether this is the best way to use it or we should use other sources

Reply:
To review today with Mostafa.

Reply:
From the conversation with Mostafa: 

- We expect policy library to be used as a set of examples that customers might need to customize.
- The way we expect now to do so is via Git, copy and paste then to update with the values that better suit them.
- Library is enterprise and its value would be protected by the ww - customer agreement.

Reply:
How do people get access to it?

Reply:
To follow up next week in policy distribution out of git